daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -730,6 +722,14 @@ object Main extends Logging {
       case "json" => new JsonInfosetOutputter(os, true)
       case "jdom" => new JDOMInfosetOutputter()
       case "w3cdom" => new W3CDOMInfosetOutputter()
+      case "sax" => {
+        // we don't actually use this infoset outputter object we create below
+        // for anything other than associating the "sax" option with the
+        // SAXInfosetOutputter. The actually SAXInfosetOutputter is contained
+        // within the DaffodilXMLReader object (see newXMLReaderInstance) of
+        // Dataprocessor
+        new SAXInfosetOutputter(new SAXHandler())
+      }
```

Comment:
So it looks like I missed something on rebase, we actually do use this in the outputter match when we're outputting the result of the parse. 

                  case _: SAXInfosetOutputter => {
                    new org.jdom2.output.XMLOutputter().output(saxContentHandler.getDocument, output)
                  }

So I updated the above to specify a null content handler, so any attempts to use it will reult in a nullpointerexception but we can still use it for a match. Would that be sufficient?

        // we don't actually use this infoset outputter object we create below
        // for anything other than associating the "sax" option with the
        // SAXInfosetOutputter. So the contenthandler is set to null, and any 
        // attempts to use this will result in a NullPointerException. 
        // 
        // The actual SAXInfosetOutputter is contained
        // within the DaffodilXMLReader object (see newXMLReaderInstance) of
        // Dataprocessor
        new SAXInfosetOutputter(null)

Reply:
I'd prefer to not create a SAXInfosetOutputter that's never actually used aside from a match case. Feels fragile since someone could accidentally use it thinking it has the output of the sax result.

Perhaps we should rethink this--relying on NPE for code correctness is probably a bad idea and I shouldn't have suggested it in the first place.

Really, we shouldn't call getInfosetOutputter at all if we aren't going to use the result. Perhaps we use something like an ``Either``, so we either have an ``InfosetOutputter``, or a ``ContentHandler``? So the code becomes something like:

```scala
val outputter: Either[InfosetOutputter, ContentHandler] =
  if (infosetType == "sax") Right(new SAXHandler())
  else Left(getInfosetOutputter(...))
```

Now later logic can use Left vs Right to determine whether to use the new SAX API or the old API, for example:
```scala
val parseResult = outputter match {
  case Left(io: InfosetOutputter) => // use io with the old API
  csae Right(ch: ContentHandler) => // use ch with the new sax API
}

 And it can also be used to match when outputting the infoset as well, something like:
```scala
output match {
  case Left(sxsml: ScalaXMLInfosetOutputtter) => ...
  case Left(jdom: JDOMInfosetOutputter) => ...
  case Right(saxHandler: SAXHandler) => ...
  case _ => // do nothing
}
```

Reply:
Oh cool, I've never used Either before. Will do. Thanks!